File: Data Structures & Algorithms/divide-array-into-equal-pairs/submission-0.py
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        numMap = {}
        for n in nums:
            if n not in numMap.keys():
                numMap[n] = 1
            else:
                numMap[n] += 1
        for key in numMap.keys():
            if numMap[key] % 2 != 0:
                return False

        return True

File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    numMap := make(map[int]int)
	for i, val := range nums {
		diff := target - val
		if index, ok := numMap[diff]; ok {
			return []int{index, i}
		}
		numMap[val] = i
	}
	return []int{}
}
